src/createSlice.ts

```diff
@@ -172,35 +183,40 @@ export function createSlice<
   }
   const reducers = options.reducers || {}
   const extraReducers = options.extraReducers || {}
-  const actionKeys = Object.keys(reducers)
-
-  const reducerMap = actionKeys.reduce((map, actionKey) => {
-    let maybeEnhancedReducer = reducers[actionKey]
-    map[getType(name, actionKey)] =
-      typeof maybeEnhancedReducer === 'function'
-        ? maybeEnhancedReducer
-        : maybeEnhancedReducer.reducer
-    return map
-  }, extraReducers)
-
-  const reducer = createReducer(initialState, reducerMap)
-
-  const actionMap = actionKeys.reduce(
-    (map, action) => {
-      let maybeEnhancedReducer = reducers[action]
-      const type = getType(name, action)
-      map[action] =
-        typeof maybeEnhancedReducer === 'function'
-          ? createAction(type)
-          : createAction(type, maybeEnhancedReducer.prepare)
-      return map
-    },
-    {} as any
-  )
+  const reducerNames = Object.keys(reducers)
+
+  const sliceCaseReducersByName: SliceCaseReducers<State, any> = {}
+  const sliceCaseReducersByType: SliceCaseReducers<State, any> = {}
+  const actionCreators: any = {}
```

Comment:
The `actionCreators : any` here was actually how the existing code worked as far as I can see, and the types of the exported action creators are still correct based on VS Code's inspection and the type tests.

Reply:
As the method has a defined return type, as long as the value you are returning somehow satisfies that type, you can do pretty much inside - the outside interface is defined. 
Even casting to any would be valid here, and in some places I think it is to make stuff less difficult. 

As an alternative, for example you could type the `actionCreators` to `CaseReducerActions<CaseReducers>`. As a result, the assignment of `{}` would fail, and you would have to assign it `{} as any` instead. 
But you've got to die one death here, and I think that would be acceptable, as it would at least guard all assignments. (Even though it could never make sure all the values were actually assigned, just that those being assigned are correct. But still, more typesafe than `any` alone)

But, as I said, even typing this as `any` as it is is probably okay - as long as the method is tested thoroughly, and that it is.